repos/resource_git_credential.go

```diff
@@ -0,0 +1,128 @@
+type GitCredentialsAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+// GitCredentialsAPI creates GitCredentialsAPI instance from provider meta
+func NewGitCredentialsAPI(ctx context.Context, m interface{}) GitCredentialsAPI {
+	return GitCredentialsAPI{m.(*common.DatabricksClient), ctx}
+}
+
+type GitCredentialRequest struct {
+	PAT      string `json:"personal_access_token" tf:"alias:access_token"`
+	Provider string `json:"git_provider"`
+	UserName string `json:"git_username" tf:"alias:user_name"`
+}
+
+type GitCredentialResponse struct {
+	ID       int64  `json:"credential_id"`
+	UserName string `json:"git_username"`
+	Provider string `json:"git_provider"`
+}
+
+type GitCredentialList struct {
+	Credentials []GitCredentialResponse `json:"credentials,omitempty"`
+}
+
+// ID returns job id as string
+func (r GitCredentialResponse) GitCredentialID() string {
+	return fmt.Sprintf("%d", r.ID)
+}
+
+func (a GitCredentialsAPI) Delete(id string) error {
+	return a.client.Delete(a.context, fmt.Sprintf("/git-credentials/%s", id), nil)
+}
+
+func (a GitCredentialsAPI) List() ([]GitCredentialResponse, error) {
+	resp := GitCredentialList{}
+	err := a.client.Get(a.context, "/git-credentials", nil, &resp)
+	return resp.Credentials, err
+}
+
+func (a GitCredentialsAPI) Read(id string) (resp GitCredentialResponse, err error) {
+	err = a.client.Get(a.context, fmt.Sprintf("/git-credentials/%s", id), nil, &resp)
+	return
+}
+
+func (a GitCredentialsAPI) Create(req GitCredentialRequest) (resp GitCredentialResponse, err error) {
+	err = a.client.Post(a.context, "/git-credentials", &req, &resp)
+	return
+}
+
+func (a GitCredentialsAPI) Update(id string, req GitCredentialRequest) error {
+	return a.client.Patch(a.context, fmt.Sprintf("/git-credentials/%s", id), &req)
+}
+
+func ResourceGitCredential() *schema.Resource {
+	s := common.StructToSchema(GitCredentialRequest{}, func(s map[string]*schema.Schema) map[string]*schema.Schema {
+		s["git_provider"].DiffSuppressFunc = func(k, old, new string, d *schema.ResourceData) bool {
+			return strings.EqualFold(old, new)
+		}
+		s["force"] = &schema.Schema{
+			Type:     schema.TypeBool,
+			Optional: true}
+		return s
+	})
+
+	return common.Resource{
+		Schema:        s,
+		SchemaVersion: 1,
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			api := NewGitCredentialsAPI(ctx, c)
+			var req GitCredentialRequest
+			common.DataToStructPointer(d, s, &req)
+			resp, err := api.Create(req)
+
+			if err != nil {
+				if !d.Get("force").(bool) || !strings.HasPrefix(err.Error(), "Only one Git credential is supported at this time") {
+					return err
+				}
+				creds, err := api.List()
+				if err != nil {
+					return err
+				}
+				if len(creds) != 1 {
+					return fmt.Errorf("list of credentials is either empty or have more than one entry (%d)", len(creds))
+				}
+				resp = creds[0]
+				log.Printf("[INFO] new resp=%v", resp)
```

Comment:
add a context, e.g. `updating existing git credentials` or something. or remove it.